### PR TITLE
CI: add a cargo deny job

### LINKED ISSUE #1178: ci: Run cargo-deny on the CI (regularly) and maintain a deny.toml
This will make sure that we don't have duplicate dependencies in different versions, only depend on crates with suitable licenses (i.e. non-copyleft OSI licenses here) and also don't depend on anything that ships with dubious binaries (hello serde-derive).

In gstreamer-rs / gst-plugins-rs I'm already using that since a few years and it's very useful.

I don't know what Github Actions offers here but I guess someone already wrote an extension or even bot (like dependabot) for this.

CC @bilelmoussaoui @GuillaumeGomez

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Fixes #1178

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -138,6 +138,12 @@ jobs:
           command: fmt
           args: --all -- --check
 
+  cargo-deny:
+    runs-on: ubuntu-latest
```
Comment (sdroege) [inline .github/workflows/CI.yml:142]: Is it possible to in addition run this every day or once a week or so?

Comment (bilelmoussaoui) [reply .github/workflows/CI.yml:142]: It should be yes, I will look into it. But not sure if we will receive some form of notification when it fails

Comment (sdroege) [reply .github/workflows/CI.yml:142]: I get a mail whenever a build on `master` fails, so I would expect that to happen. To be checked :)

Comment (sdroege) [reply .github/workflows/CI.yml:142]: Should we just merge this for now and worry about the scheduled job independently? I don't know how hard that is to add.

Comment (bilelmoussaoui) [reply .github/workflows/CI.yml:142]: It should be easy, I just didn't had the time for it yet. Hopefully this weekend

Comment (sdroege) [reply .github/workflows/CI.yml:142]: Let's merge it as-is for now then? @bilelmoussaoui

Comment (bilelmoussaoui) [reply .github/workflows/CI.yml:142]: yeah, sounds good to me

**Pull request merged**